sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -749,9 +749,11 @@ case class SubqueryExec(name: String, child: SparkPlan)
   private lazy val relationFuture: Future[Array[InternalRow]] = {
     // relationFuture is used in "doExecute". Therefore we can get the execution id correctly here.
     val executionId = sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY)
+    val localProps = Utils.cloneProperties(sparkContext.getLocalProperties)
```

Comment:
Shouldn't we fix this in `SQLExecution.withExecutionId`? That way we can eliminate the issue you are also trying to fix in https://github.com/apache/spark/pull/27267.

Reply:
When it execute the Future object in `SQLExecution.withExecutionId` the thread from the pool has already lost the local properties so it wont work

Reply:
Fair point. I prefer creating a narrow waist here. How about you add a method to SQLExecution that creates the future and sets up the local properties?

Reply:
Yes, seems good approach. let me try to modify as per your suggestion